pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -69,8 +69,9 @@ func ApplyLocalityLBSetting(
 	// one of Distribute or Failover settings can be applied.
 	if localityLB.GetDistribute() != nil {
 		applyLocalityWeight(locality, loadAssignment, localityLB.GetDistribute())
-	} else if enableFailover {
 		// Failover needs outlier detection, otherwise Envoy will never drop down to a lower priority.
+		// Do not apply default failover when locality LB is disabled.
+	} else if enableFailover && (localityLB.Enabled == nil || localityLB.Enabled.Value) {
```

Comment:
We donot really need `ENabled` check any more,  users would complain why I have already defined localityLB setting, it still not work.

Reply:
We should come to a consensus, no one would make trouble to set it when they really donot need it.

Reply:
But how will user disable auto failover with out that flag?

Reply:
https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer_test.go#L161 - Also look at this test.

If your users given conflicting values like enabled false and provide some LocalityLBSetting, we will ignore LocalityLBSetting and it is users problem.

Since that field is there, we should apply it consistently

Reply:
Personally  i think it only make sense when user want to disable mesh settings

Reply:
Yes, the check logic here is right

Reply:
It is actually does that also - If the destination rule has disabled it, it honours and does not apply failover rules for that cluster endpoints.
This code change enables that behaviour - without this, even if locality lb is disabled at mesh level/destination rule level, default failover is applied.